src/v/cloud_storage/remote_segment.cc

```diff
@@ -361,57 +519,62 @@ ss::future<> remote_segment::run_hydrate_bg() {
                 // and retrieve the file out of it or hydrate.
                 // If _data_file is initialized we can use it safely since the
                 // cache can't delete it until we close it.
-                auto status = co_await _cache.is_cached(_path);
+                auto tx_path = generate_remote_tx_path(_path);
+                auto segment_status = co_await _cache.is_cached(_path);
+                auto txrange_status = co_await _cache.is_cached(tx_path);
+                auto status = combine_statuses(segment_status, txrange_status);
                 switch (status) {
-                case cache_element_status::in_progress:
+                case segment_txrange_status::in_progress:
```

Comment:
if segment is not available, but tx range is in progress, then we don't hydrate the segment. Is it expected?